src/fu-engine.c

```diff
@@ -4512,6 +4512,23 @@ fu_engine_md_refresh_device_name_category(FuEngine *self, FuDevice *device, XbNo
 		fu_device_remove_internal_flag(device,
 					       FU_DEVICE_INTERNAL_FLAG_MD_SET_NAME_CATEGORY);
 	}
+
+	/* using GUIDs to set internal flag ignore-system-power doesn't scale well */
+	if (fu_device_has_vendor_id(device, "DMI:LENOVO")) {
```

Comment:
why restrict this to Lenovo?

Reply:
I guess what i'm trying to do is restrict it to laptop batteries, but we don't have enough metadata to know the difference between a battery in a laptop and a battery in a wireless mouse for instance.

Reply:
you could just only do it with the uefi-capsule plugin instead then perhaps?

Reply:
Yup, that's a good heuristic I guess. Fixed.